depends/packages/expat.mk

```diff
@@ -1,15 +1,18 @@
 package=expat
-$(package)_version=2.4.1
+$(package)_version=2.4.8
 $(package)_download_path=https://github.com/libexpat/libexpat/releases/download/R_$(subst .,_,$($(package)_version))/
 $(package)_file_name=$(package)-$($(package)_version).tar.xz
-$(package)_sha256_hash=cf032d0dba9b928636548e32b327a2d66b1aab63c4f4a13dd132c2d1d2f2fb6a
+$(package)_sha256_hash=f79b8f904b749e3e0d20afeadecf8249c55b2e32d4ebb089ae378df479dcaf25
 
+# -D_DEFAULT_SOURCE defines __USE_MISC, which exposes additional
+# definitions in endian.h, which are required for a working
+# endianess check in configure when building with -flto in Guix.
```

Comment:
nit: Could be improved basing in this [comment](https://github.com/bitcoin/bitcoin/pull/25697#pullrequestreview-1050173051).

Reply:
Adjusted the comment to remove the mention of Guix.